adapter/internal/notifier/deployment_notifier.go

```diff
@@ -82,3 +87,52 @@ func SendRevisionUpdate(deployedRevisionList []*DeployedAPIRevision) {
 		}
 	}
 }
+
+// SendRevisionUndeploy - send the undeployed revision information to control plane
+func SendRevisionUndeploy(apiUUID string, revisionUUID string, environment string) {
+	conf, _ := config.ReadConfigs()
+	cpConfigs := conf.ControlPlane
+
+	revisionEP := cpConfigs.ServiceURL
+	if strings.HasSuffix(revisionEP, "/") {
+		revisionEP += unDeployedRevisionEP
+	} else {
+		revisionEP += "/" + unDeployedRevisionEP
+	}
+
+	if apiUUID == "" || revisionUUID == "" || environment == "" {
+		return
+	}
+
+	removedRevision := UnDeployedAPIRevision{
+		APIUUID:      apiUUID,
+		RevisionUUID: revisionUUID,
+		Environment:  environment,
+	}
+
+	jsonValue, _ := json.Marshal(removedRevision)
+	basicAuth := authBasic + auth.GetBasicAuth(cpConfigs.Username, cpConfigs.Password)
+	retries := 0
+	for retries < 3 {
+		retries++
+		req, _ := http.NewRequest("POST", revisionEP, bytes.NewBuffer(jsonValue))
+		req.Header.Set(authHeader, basicAuth)
+		req.Header.Set(contentTypeHeader, "application/json")
+		resp, err := tlsutils.InvokeControlPlane(req, cpConfigs.SkipSSLVerification)
+
+		success := true
+		if err != nil {
+			logger.LoggerNotifier.Warnf("Error response from %v for attempt %v : %v", revisionEP, retries, err.Error())
```

Comment:
Isn't it better to set it as an error log?

Reply:
+1 and 
Fixed via : https://github.com/wso2/product-microgateway/pull/2503/commits/4d77d188a4aefb79e5d8582c06613d46ae0a1348